### PR TITLE
8435 zpool.1m and zfs.1m: minor cleanup 

### PR DESCRIPTION
**Pull request by gmelikov**
This commit is a result of re-read while porting zpool.1m to ZFSonLinux.

See https://github.com/zfsonlinux/zfs/commit/cda0317e4d2a1277b328e4fc42ee3699bbe46c12

Signed-off-by: George Melikov <mail@gmelikov.ru>
Closes #8435
Closes #3796

### CODE REVIEW

File: usr/src/man/man1m/zpool.1m [hunk 1]
```diff
@@ -1733,7 +1736,7 @@ The failed device can be replaced using the following command:
 .Ed
 .Pp
 Once the data has been resilvered, the spare is automatically removed and is
-made available should another device fails.
+made available for use should another device fails.
```
Comment (yuripv) [inline usr/src/man/man1m/zpool.1m]: s/fails/fail/ too.

File: usr/src/man/man1m/zpool.1m [hunk 2]
```diff
@@ -77,7 +78,7 @@
 .Nm
 .Cm import
 .Op Fl D
-.Op Fl d Ar dir
+.Op Fl c Ar cachefile Ns | Ns Fl d Ar dir
```
Comment (yuripv) [inline usr/src/man/man1m/zpool.1m]: I'm not sure if "-c cachefile" was left out on purpose here.  If that's not the case, please make sure to update this form of "import" subcommand in the DESCRIPTION as well.

File: usr/src/man/man1m/zpool.1m [hunk 3]
```diff
@@ -660,7 +661,7 @@ to the enabled state.
 See
 .Xr zpool-features 5
 for details on feature states.
-.It Sy listsnaps Ns = Ns Sy on Ns | Ns Sy off
+.It Sy listsnapshots Ns = Ns Sy on Ns | Ns Sy off
```
Comment (yuripv) [inline usr/src/man/man1m/zpool.1m]: Could please include the https://www.illumos.org/issues/3796 in the commit message for this change.

File: usr/src/man/man1m/zpool.1m [hunk 4]
```diff
@@ -1367,7 +1370,7 @@ Upon reboot, the specified physical device reverts to its previous state.
 .Ar pool Ar device Ns ...
 .Xc
 Brings the specified physical device online.
-This command is not applicable to spares.
+This command is not applicable to spares or cache devices.
```
Comment (yuripv) [inline usr/src/man/man1m/zpool.1m]: I don't think this is correct:
```
antares:root:~# zpool create data c3t0d0 cache c3t1d0
antares:root:~# zpool offline data c3t1d0
antares:root:~# zpool status data
  pool: data
 state: ONLINE
status: One or more devices has been taken offline by the administrator.
        Sufficient replicas exist for the pool to continue functioning in a
        degraded state.
action: Online the device using 'zpool online' or replace the device with
        'zpool replace'.
  scan: none requested
config:

        NAME        STATE     READ WRITE CKSUM
        data        ONLINE       0     0     0
          c3t0d0    ONLINE       0     0     0
        cache
          c3t1d0    OFFLINE      0     0     0

errors: No known data errors
antares:root:~# zpool online data c3t1d0
antares:root:~# zpool status data
  pool: data
 state: ONLINE
  scan: none requested
config:

        NAME        STATE     READ WRITE CKSUM
        data        ONLINE       0     0     0
          c3t0d0    ONLINE       0     0     0
        cache
          c3t1d0    ONLINE       0     0     0

errors: No known data errors
antares:root:~#
```

**Comment by gmelikov** [comment]
@yuripv thanks, fixed.

zfs.1m added.

File: usr/src/man/man1m/zfs.1m [hunk 5]
```diff
@@ -3145,31 +3147,35 @@ property.
 The following permissions are available:
 .Bd -literal
 NAME             TYPE           NOTES
-allow            subcommand     Must also have the permission that is being
```
Comment (gmelikov) [inline usr/src/man/man1m/zfs.1m]: This line and above were changed after @yuripv review, formatted output doesn't wrap nicely on an 80 column terminal.

File: usr/src/man/man1m/zpool.1m (hunk 2 above)
Comment (gmelikov) [inline usr/src/man/man1m/zpool.1m]: @yuripv @ahrens thanks for clarification, I removed it.

File: usr/src/man/man1m/zpool.1m [hunk 6]
```diff
@@ -1011,6 +1014,7 @@ performed.
 .Cm import
 .Op Fl D
 .Op Fl d Ar dir
+.Op Fl c Ar cachefile Ns | Ns Fl d Ar dir
```
Comment (ahrens) [inline usr/src/man/man1m/zpool.1m]: Did you mean to remove the `-c cachefile` here too?

Comment (gmelikov) [inline usr/src/man/man1m/zpool.1m]: @ahrens thanks, removed it too.

**Comment by ahrens** [comment]
The failed tests don't seem to have anything to do with manpages, so I'm going to assume that they are not related to your changes.  @prakashsurya do we need to mask these off, or investigate?
```
    FAIL cli_root/zpool_create/zpool_create_008_pos (expected PASS)
    FAIL cli_root/zpool_remove/zpool_remove_001_neg (expected PASS)
```

**Comment by prakashsurya** [comment]
I saw those fail in the most recent illumos sync, too; see here: https://github.com/openzfs/openzfs/pull/431#issuecomment-317472487

I'll open a PR to mask these failures until we have a change to properly investigate.

**Comment by gmelikov** [comment]
Rebased on master.

**Pull request closed without merging**